knimin/tests/test_ag_new_kit.py

```diff
@@ -0,0 +1,129 @@
+                    ["000033916", "000033917"]]]
+        response = self.post('/ag_new_kit/download/',
+                             {'kitinfo': dumps(kitinfo),
+                              'fields': ('kit_id,password,verification_code,'
+                                         'barcodes')})
+        self.assertEqual(response.code, 200)
+        # is it necessary to unpack the zipfile and check its content?
+        self.assertEqual(response.headers['Content-Disposition'],
+                         'attachment; filename=kitinfo.zip')
+
+
+class TestAGNewKitHandler(TestHandlerBase):
+    def test_get_not_authed(self):
+        response = self.get('/ag_new_kit/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/login/?next=%s' %
+                         (port, url_escape('/ag_new_kit/')))
+
+    def test_get(self):
+        self.mock_login_admin()
+        response = self.get('/ag_new_kit/')
+        self.assertEqual(response.code, 200)
+        for project in db.getProjectNames():
+            self.assertIn("<option value='%s'>%s</option>" %
+                          (project, project), response.body)
+        self.assertIn("%i</span> unassigned barcodes" %
+                      len(db.get_unassigned_barcodes()), response.body)
+
+    def test_post(self):
+        self.mock_login_admin()
+        kits = [1, 2]
+        swabs = [2, 2]
+        tag = 'abc'
+
+        # check for correct results
+        response = self.post('/ag_new_kit/',
+                             {'tag': tag,
+                              'projects': ['PROJECT2', 'PROJECT5'],
+                              'swabs': swabs,
+                              'kits': kits,
+                              })
+        kitinfo = loads(response.body)
+        self.assertEqual(len(kitinfo['kitinfo']), sum(kits))
+        for k in kitinfo['kitinfo']:
+            self.assertIn(tag, k[0])
+        self.assertEqual(len(kitinfo['kitinfo'][0][-1]), swabs[0])
+        self.assertEqual(len(kitinfo['kitinfo'][1][-1]), swabs[1])
+        self.assertEqual(kitinfo['fields'],
+                         "kit_id,password,verification_code,barcodes")
+
+        # missing argument
+        response = self.post('/ag_new_kit/',
+                             {'projects': ['PROJECT2', 'PROJECT5'],
+                              'swabs': swabs,
+                              'kits': kits,
+                              })
+        self.assertEqual(response.code, 400)
+
+        # too long tag
+        response = self.post('/ag_new_kit/',
+                             {'tag': 'toolongtag',
+                              'projects': ['PROJECT2', 'PROJECT5'],
+                              'swabs': swabs,
+                              'kits': kits,
+                              })
+        self.assertEqual(response.code, 500)
```

Comment:
I would prefer to first complete tests for the existing - maybe crappy - code base and only start improving it afterwards, as we already decided for issue #102